src/bitcoin-cli.cpp

```diff
@@ -103,8 +103,31 @@ static int AppInitRPC(int argc, char* argv[])
         fprintf(stderr, "Error: Specified data directory \"%s\" does not exist.\n", GetArg("-datadir", "").c_str());
         return EXIT_FAILURE;
     }
+    // For bitcoin-cli, the -wallet parameter can only be specified from the
+    // command line, not in the conf file. This is because the conf file can
+    // contain multiple -wallet args, so it's not possible to determine which
+    // one to use.  Furthermore, if bitcoin-cli used the -wallet arg from the
+    // conf file, we'd get 'trapped' in wallet mode and not be able to issue
+    // commands to the node.
+    //
+    // Store the command line -wallet arg here so we can restore it after
+    // parsing the conf file.
+    bool use_wallet = false;
+    std::string wallet_name;
+    if (gArgs.GetArgs("-wallet").size() > 1) {
+        fprintf(stderr, "Error: Only one -wallet option may be used.\n");
+        return EXIT_FAILURE;
+    } else if (gArgs.GetArgs("-wallet").size() == 1) {
+        wallet_name = (gArgs.GetArgs("-wallet"))[0];
+        use_wallet = true;
+    }
     try {
         ReadConfigFile(GetArg("-conf", BITCOIN_CONF_FILENAME));
+        // restore the command line -wallet argument
+        gArgs.ClearArg("-wallet");
```

Comment:
I prefer not to set this as a global or pass it through `main()`, `CommandLineRPC()` and `CallRPC()`. It makes sense to me to set up all the config in `AppInitRPC()` and have it contained in the standard `gArgs` global.

Happy to reconsider if making changes to util.cpp is a blocker for this PR.